### PR TITLE
docs(general): provide oop example that matches cli output and fix functional example

### PR DESCRIPTION
**Pull request by thomas-w-grant**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/elf/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Documentation for the repository pattern has the current behavior:
1.  Functional example calls `createStore` with `{name}` with no name set.  
2. OOP acronym is not defined.  Learner bias that everyone knows that OOP is Object-oriented programming.  Most do, but such little effort to serve all.  ❤️ 
3. OOP example is not complete:  Instead, you get `// ...`.  Approach bias.  Functional fans get a complete example.  OOP fans do not.    Not friendly to your Angular friends.
4. OOP example does not match the awesome output produced by the CLI when `repoTemplate` option is set to `class` for `repo` command.

Issue Number: N/A

## What is the new behavior?

Documentation for the repository pattern now has the following behavior:
1. Functional example calls `createStore` with `{name: 'auth'}`.
2. OOP acronym is defined once with the following text: "object-oriented programming (OOP)"
3. OOP example is equally complete as the functional example.
4. OOP example now matches the awesome output produced by the CLI.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: docs/docs/repository.mdx [hunk 1]
```diff
@@ -26,21 +29,45 @@ The Repository pattern provides 2 main benefits:
 1. Using the pattern, you can replace your data store without changing your business code.
 2. It encourages you to implement all store operations in one place, making your code more reusable and easy to find.
 
-You can also use the `OOP` approach:
+If working in Angular, you can also use the object-oriented programming (OOP) approach:
```
Comment (NetanelBasal) [inline docs/docs/repository.mdx:32]: This shouldn't be specific to Angular.

File: docs/docs/repository.mdx [hunk 2]
```diff
@@ -26,21 +29,45 @@ The Repository pattern provides 2 main benefits:
...
+  public updateUser(user: AuthProps['user']) {
+    this.store.update((state) => {
+      return {
+        ...state,
+        user,
+      };
+    });
+  }
+
+  private createStore(): typeof store {
```
Comment (NetanelBasal) [inline docs/docs/repository.mdx:64]: This style will be used only if you set the [inlineStoreInClass](https://ngneat.github.io/elf/docs/cli#config) to `true`.

Comment (NetanelBasal) [reply docs/docs/repository.mdx:64]: We can add a new block showing this example.

File: docs/docs/repository.mdx (hunk 1 above)
Comment (thomas-w-grant) [reply docs/docs/repository.mdx:32]: 👍🏻 I definitely second guessed myself when I wrote that.  I will remove it.

I based my decision to include it on seeing this in the CLI documentation:

image

File: docs/docs/repository.mdx (hunk 2 above)
Comment (thomas-w-grant) [reply docs/docs/repository.mdx:64]: You got me thinking.  Should these new examples go in the repository documentation or the CLI documentation.  I could make an argument for both.  What are your thoughts?

Some feedback, the team that I work on is considering using Elf but was very concerned about the lack of Angular examples.  I told them that the CLI produces really great Angular code (in my opinion (IMO)).  Their feedback was that they wished the documentation showed it.

Comment (NetanelBasal) [reply docs/docs/repository.mdx:64]: We can open a dedicated section of "Angular Examples"

Comment (thomas-w-grant) [reply docs/docs/repository.mdx:64]: Oh .... I like that.  ❤️

**Comment by thomas-w-grant** [comment]
I have removed the mention of Angular and I have removed the example.  Perhaps it will appear in the near future in a "Angular examples" section.  😄 🙏🏻 ❤️

**Pull request merged**